### PR TITLE
Add a __len__() function to doubly_linked_list.py

### PR DESCRIPTION
**Pull request by Lakshmikanth2001**
Blocked by #2052

### **Describe your change:**
in linked list class constructor i have added a data member called size and a function call length

* [x] Add an algorithm?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/linked_list/__init__.py [hunk 1]
```diff
@@ -6,17 +6,22 @@ def __init__(self, item, next):
...
             return None
         else:
             item = self.head.item
             self.head = self.head.next
+            self.size-=1
             return item
 
     def is_empty(self):
         return self.head is None
+    def size(self):
```
Comment (cclauss) [inline data_structures/linked_list/__init__.py:26]: Please define [`__len__()`](https://docs.python.org/3/reference/datamodel.html#object.__len__) instead.

Is it better to maintain a `size` variable or is better to visit all nodes and count them?

**Comment by cclauss** [comment]
Please resolve conflicting files.

File: data_structures/linked_list/__init__.py (hunk 1 above)
Comment (Lakshmikanth2001) [reply data_structures/linked_list/__init__.py:26]: Sir I think maintaining size variable is better

**Comment by onlinejudge95** [comment]
@Lakshmikanth2001 please resolve the conflict

**Comment by cclauss** [comment]
Blocked by #2052

**Comment by cclauss** [comment]
@Lakshmikanth2001 Thanks for this pull request because it found a flaw in doubly_linked_list.py.  It would be great is you could fix that algorithm so that the tests in #2052 pass.  If not, I will try to fix that at the end of this week.  Thanks again.

**Pull request closed without merging**